Dynatrace/dynatrace-operator
# Change stored version to v1beta3

## Description

This PR has all changes requires to use stored version as  `v1beta3` including:

- Replace `"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta2/dynakube"` to `"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube"` mostly everywhere, except few files;
- move conversion logic to `v1beta3` package;
- 

## How can this be tested?

Make sure we store every CR (dynakube) using `v1beta3`, some testing ideas:
- Apply `dynakube v1beta1` -> stored as `v1beta3`
- Apply `dynakube v1beta2` -> stored as `v1beta3`
- Apply `dynakube v1beta3` -> stored as `v1beta3`

**Optional:**
- Install the previous release, add `dynakube v1beta2`, and then upgrade the operator using this PR.
